### DISCUSSION TITLE
RSS3 Protocol v0.4.0 Adventure: Phase II

### DISCUSSION THREAD

**Discussion by DIYgod** [NaturalSelectionLabs/RSS3-Protocol#32] [Ideas]
![](https://rss3.io/assets/logo.dc94fdc9.gif)

After [the last enthusiastic discussion](https://github.com/NaturalSelectionLabs/RSS3-Protocol/discussions/25), we are happy to announce that we have completed the design and documentation for protocol v0.4.0-rc.1 and a series of RSS3 Improvement Proposals.

Before the adventure begins, I would like to answer questions that was frequently asked in the last discussion

1. Channel/community feature: It is a very complex feature and we expect it to be implemented through a reserved `controller` field, but we are not sure how to implement it now and we will have a future discussion dedicated to this issue.
2. NFT PFP: It is easy to implement in our design, just set the value of the `avatar` field to your asset URI.
3. User Relationship Maintenance: We have opened a dedicated developer communication channel on Discord's #ask-the-developers channel and plan to organize regular (weekly?) text AMA events.
4. IM feature: It has a big gap with the function of information aggregation, and I'm afraid we can't implement this function for the time being, but we will keep thinking about this issue.
5. Privacy: We have added filters to items to filter the content that the user does not want to display.
6. More Supported Networks/Data: When the development of [RSS3 Pregod](https://github.com/NaturalSelectionLabs/RSS3-PreGod) (implementation program of v0.4.0) is complete, we will open up the index modules for contributions, and then everyone can submit their favourite content to the network.
7. Custom fields: We realized that custom fields might be a bad design, so custom fields were temporarily removed.
8. Recommendation and search: We don't reject recommendation algorithms, we just think they should be transparent and optional. We have already conducted research on recommendation algorithms and will experiment with them in Revery soon; the search function will be launched after Pregod is developed.
9. Auto items management: Currently only the filtering function is designed, the grouping function needs to be discussed subsequently.
10. Censorship: We expect to implement the censorship function independently of the RSS3 network, and applications can choose which one to use.
11. Identity verification: Not currently under consideration.
12. Protocol boundary: We've kept the concepts of protocol and network strictly separate.

Through the last discussion, we realized that we had no way to stop people from discussing things beyond the scope of the protocol, but I still strongly suggest that we only discuss the protocol here and leave the other questions for the upcoming AMA.

Let's get the adventure started!

The scope of this adventure:
- [RSS3 Protocol v0.4.0-rc.1](https://rss3.io/protocol/v0.4.0-rc.1.html)
- RSS3 Improvement Proposals
    - [RIP-1: Registered Account Platforms](https://rss3.io/protocol/RIPs/RIP-1.html)
    - [RIP-2: Registered Name Services](https://rss3.io/protocol/RIPs/RIP-2.html)
    - [RIP-3: Registered Item Networks](https://rss3.io/protocol/RIPs/RIP-3.html)
    - [RIP-4: Registered Indexed Items](https://rss3.io/protocol/RIPs/RIP-4.html)
    - [RIP-5: Registered Link Types](https://rss3.io/protocol/RIPs/RIP-5.html)
- [RSS3-PreGod](https://github.com/NaturalSelectionLabs/RSS3-PreGod)

By the way, developers who participate in meaningful discussions will also be rewarded with a special NFT to be used in future DAO.

**Comment by EdwardWason** [comment]
Whether there is a lack of an information reminder function, if there is an update on the topic of interest or ens, there should be a dapp reminder function and count statistics

**Reply by DIYgod** [reply]
Good idea, we may need to implement a webhook function on PreGod

**Comment by dearck1** [comment]
同意，浏览了一下这次调整后的框架基本把前期大家提到的痛点问题解决了，期待尽快的上线运行

**Comment by dgaxiola** [comment]
Since I've spent most of the last 2 weeks working on an app that exclusively uses custom items, I'm a bit dismayed that it's going away. Will I need to migrate to item attachments when 0.4.0 is released? I'm really looking for the Web3 decentralized equivalent of posting an RSS or Atom file to a web server, not necessarily integrate a bespoke social network into RSS3. Longer term, would it be better to host items in a traditional centralized location and build an indexer for replication into RSS3? I just want to understand the vision for content that isn't directly on a supported blockchain or a social media service that gets an indexer built by the RSS3 developers.

**Reply by DIYgod** [reply]
Don't worry, the custom item is not removed, it is a very important feature for our future, see [#Item](https://rss3.io/protocol/v0.4.0-rc.1.html#_4-5-item), what is removed is the Extensions in v0.3.1

**Comment by Alexchen359** [comment]
I have heard about chree.io will make some changes in the future. Any more details are released? I am very curious about it, chree.io could be built into a more connected and social function platform.

**Comment by ENZO7109** [comment]
Although registered with RSS3 in the early days, for a beginner, there is still a lack of understanding of many functions of the website. It is recommended to attach a detailed description to each link on the web page at the same time, so that beginners can master it as soon as possible.

**Comment by CruickAbby** [comment]
Looking forward to the official launch，Nice to see the progress of RSS3.

**Comment by monerochan-moe** [comment]
Would it be possible to add an api that gives you a list of the most recently updated files across the entire network?

I understand the focus is on profiles and assets, but for the sake of discover-ability it may be useful to be able to see a real-time live feed of all the activity being indexed, both auto indexes and custom items.

Perhaps the closest thing to this now is the commit history of https://github.com/NaturalSelectionLabs/RSS3-Network-Data/commits/main

**Comment by AboutRSS** [comment]
Hi. Since RSS is mentioned in RIP-1 and RIP-3, is it possible to include an example in RIP-4 about how an RSS feed transformed into, perhaps, `item.notes` in the regime of RSS3 ?

Cheers.
